package.json

```diff
@@ -21,7 +21,8 @@
     "test:mocha": "nyc --nycrc-path=test/.nycrc-mocha mocha --colors",
     "test:mocha-webpack": "nyc --nycrc-path=test/.nycrc-mocha-webpack mocha-webpack --colors --opts=test/mocha-webpack.opts",
     "lint:js": "eslint . --report-unused-disable-directives --color",
-    "lint:css": "stylelint --color \"client/**/*.css\""
+    "lint:css": "stylelint --color \"client/**/*.css\"",
+    "start:mocha": "webpack-dev-server --config=webpack.config-browser.js"
```

Comment:
This scripts section needs an overhaul.

1. Not in alphabetical order, which saddens me
2. We are using `:`s and `-` when we should be consistent.

Which do we want to use? I like `:` but we should change `start-dev` to `start:dev` in that case.

However, that's not for this PR. What is for this PR is ... I'm not convinced about this name.

It should probably be a `test:` prefix, not `start:`, I think. And I would call it something like `test:watch` or `test:browser` or something similar.

Reply:
Well, I mean I'm just not a huge fan of using the `start:` prefix for something that isn't running the application, you know?

> The issue with test:* is that it will get picked by the test script :/

The test script is literally just running 2 test scripts, so we can just change it to use those 2. It's not like we are likely to end up having more scripts that run here very often, and it's not like people are having to type out the long command each time, we have it coded in the package.json so it can be verbose if needs be.

I'd rather that, and call the new one `test:` than `start:`.

Fwiw, I also think the existing test scripts would probably be better as `test:client` and `test:server`, because the client tests (although they use webpack) also use mocha, and we don't care what they are using, the important thing is what we are testing.

If I want to run just 1 of the test scripts, I need to try and remember what the initial tool we use to run them is, rather than whether they are client or server tests, which confuses things.

So, I would suggesting `test:server`, `test:client`, `test:watch`, and change `test` to hard code `test:server` and `test:client` instead of `test:*`. I think that would be the cleanest and most understandable solution.




As a side note, xpaw was complaining about changing `start-dev` to `start:dev` as `:` is harder to type. This led me to think ... why do we need the `start:` prefix for dev at all? Why not just call it `dev` and we just leave `start` for the production run.

Reply:
> Well, I mean I'm just not a huge fan of using the `start:` prefix for something that isn't running the application, you know?

Yep, gotcha!

> The test script is literally just running 2 test scripts, so we can just change it to use those 2.

Done! (Also changed the `coverage` script)

> I'd rather that, and call the new one `test:` than `start:`.

Done!

> Fwiw, I also think the existing test scripts would probably be better as `test:client` and `test:server`

Done!

> So, I would suggesting `test:server`, `test:client`, `test:watch`, and change `test` to hard code `test:server` and `test:client` instead of `test:*`. I think that would be the cleanest and most understandable solution.

Done! (Except I went with `test:browser` instead of `test:watch` because `test:watch`/`watch:test` is usually used as "re-run `test` at every file changed", see https://mochajs.org/#-w---watch, whaddaya think?)

> Why not just call it `dev` and we just leave `start` for the production run.

Done!

You see, I can be a reasonable person sometimes! 😁
Happy to change any further if there are other things you'd like changed.